From: "Levin, Alexander (Sasha Levin)" <alexander.levin@verizon.com>
Date: Thu, 25 May 2017 12:58:55 +0000
Subject: tools/lib/lockdep: Add empty __is_[module,kernel]_percpu_address
Patch-mainline: v4.13-rc1
Git-commit: 3156cbaeda991f6b330dd8260d863bfad71b358e
References: bsc#1109837

This would fix the build error caused by:

	383776fa7 ("locking/lockdep: Handle statically initialized
PER_CPU locks properly")

Signed-off-by: Sasha Levin <alexander.levin@verizon.com>
Cc: Linus Torvalds <torvalds@linux-foundation.org>
Cc: Peter Zijlstra <peterz@infradead.org>
Cc: Thomas Gleixner <tglx@linutronix.de>
Cc: a.p.zijlstra@chello.nl
Cc: ben@decadent.org.uk
Link: http://lkml.kernel.org/r/20170525130005.5947-19-alexander.levin@verizon.com
Signed-off-by: Ingo Molnar <mingo@kernel.org>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 tools/lib/lockdep/uinclude/linux/lockdep.h |    5 +++++
 tools/lib/lockdep/uinclude/linux/module.h  |    5 +++++
 2 files changed, 10 insertions(+)

--- a/tools/lib/lockdep/uinclude/linux/lockdep.h
+++ b/tools/lib/lockdep/uinclude/linux/lockdep.h
@@ -51,4 +51,9 @@ static inline int debug_locks_off(void)
 #define debug_show_all_locks()
 extern void debug_check_no_locks_held(void);
 
+static __used bool __is_kernel_percpu_address(unsigned long addr, void *can_addr)
+{
+	return false;
+}
+
 #endif
--- a/tools/lib/lockdep/uinclude/linux/module.h
+++ b/tools/lib/lockdep/uinclude/linux/module.h
@@ -3,4 +3,9 @@
 
 #define module_param(name, type, perm)
 
+static inline bool __is_module_percpu_address(unsigned long addr, unsigned long *can_addr)
+{
+	return false;
+}
+
 #endif
